src/util.cpp

```diff
@@ -848,7 +850,12 @@ bool ArgsManager::ReadConfigFiles(std::string& error, bool ignore_invalid_keys)
         }
         // if there is an -includeconf in the override args, but it is empty, that means the user
         // passed '-noincludeconf' on the command line, in which case we should not include anything
-        if (m_override_args.count("-includeconf") == 0) {
+        bool emptyIncludeConf;
+        {
+            LOCK(cs_args);
```

Comment:
`ReadConfigFiles` is done once on start. Couldn't you just take the lock at the beginning of the method and remove those scoped locks further down in the method?

Reply:
This change caused the deadlock. Now reverted to previous version :-)